src/core/meshcop/dataset.cpp

```diff
@@ -67,12 +67,11 @@ Error Dataset::Info::GenerateRandom(Instance &aInstance)
 
     Clear();
 
-    mActiveTimestamp              = 1;
-    mChannel                      = preferredChannels.ChooseRandomChannel();
-    mChannelMask                  = supportedChannels.GetMask();
-    mSecurityPolicy.mRotationTime = KeyManager::kDefaultKeyRotationTime;
-    mSecurityPolicy.mFlags        = KeyManager::kDefaultSecurityPolicyFlags;
-    mPanId                        = Mac::GenerateRandomPanId();
+    mActiveTimestamp = 1;
+    mChannel         = preferredChannels.ChooseRandomChannel();
+    mChannelMask     = supportedChannels.GetMask();
+    mSecurityPolicy  = SecurityPolicy();
```

Comment:
Continue from previous comment:

Suggest  having a method `SecurityPolicy::SetToDefault()` and use it here instead of `mSecurityPolicy  = SecurityPolicy();`?
- Makes the code more readable
- Also a bit more efficient on most toolchains. 
   - With `mSecurityPolicy  = SecurityPolicy();` some toolchains may do an extra object allocation/constructor call and then do a copy. 
   -  With `mSecurityPolicy.SetToDefault()` we know the fields of `mSecurityPolicy` are set directly (on all toolchains).

Reply:
`mSecurityPolicy` is acutally `otSecurityPolicy`, but we can define `SetToDefault` as a static method of `SecurityPolicy` and set `mSecurityPolicy` with `SecurityPolicy::SetToDefault(mSecurityPolicy);`. Updated, PTAL.

Reply:
How about doing a `static_cast` on the `mSecurityPolicy`? I thin would work and would be simpler.

We do this with all other types, like from `otIp6Address` to `Ip6::Address`, etc.

Reply:
SGTM, updated!